packages/mui-material/src/ButtonBase/ButtonBase.js

```diff
@@ -328,6 +336,7 @@ const ButtonBase = React.forwardRef(function ButtonBase(inProps, ref) {
     ...props,
     centerRipple,
     component,
+    isButton: ComponentProp === 'button',
```

Comment:
isButton semantically is wrong, `ComponentProp` could be a wrapped button.

I think:

```suggestion
    mightNotBeButton: ComponentProp !== 'button',
```

is closer to what we do here, we play it safe, we disable text selection if we have a doubt.

Reply:
`isButton` is used to turn off event callbacks with `pointer-events: none`, not turning off text selection.

Reply:
Yes, but my issue is with the semantics. I think it sends the wrong signal to have `isButton: true` when it doesn't render a `<button>`. e.g. if providing a styled('button') as a component prop.

Reply:
I removed it because of https://github.com/mui/material-ui/pull/38158#discussion_r1349310333. It's not needed anymore after that change.